models/maisi_ct_generative/scripts/find_masks.py

```diff
@@ -12,109 +12,144 @@
         top_index = top_index[0]
         bottom_index = [index for index, element in enumerate(_item["bottom_region_index"]) if element != 0]
         bottom_index = bottom_index[0]
 
-        flag = False
+        # whether to keep this mask, default to be True.
+        keep_mask = True
+
+        # if candiate mask does not contain all the body_region, skip it
         for _idx in body_region:
             if _idx > bottom_index or _idx < top_index:
-                flag = True
+                keep_mask = False
 
-        # check if candiate mask contains tumors
         for tumor_label in [23, 24, 26, 27, 128]:
             # we skip those mask with tumors if users do not provide tumor label in anatomy_list
             if tumor_label not in anatomy_list and tumor_label in _item["label_list"]:
-                flag = True
+                keep_mask = False
 
         if check_spacing_and_output_size:
-            # check if the output_size and spacing are same as user's input
+            # if the output_size and spacing are different with user's input, skip it
             for axis in range(3):
                 if _item["dim"][axis] != output_size[axis] or _item["spacing"][axis] != spacing[axis]:
-                    flag = True
+                    keep_mask = False
 
-        if flag is True:
-            continue
+        if keep_mask:
+            # if decide to keep this mask, we pack the information of this mask and add to final output.
+            candidate = {
+                "pseudo_label": os.path.join(mask_foldername, _item["pseudo_label_filename"]),
+                "spacing": _item["spacing"],
+                "dim": _item["dim"],
+                "top_region_index": _item["top_region_index"],
+                "bottom_region_index": _item["bottom_region_index"],
+            }
 
-        candidate = {}
-        if "label_filename" in _item:
-            candidate["label"] = os.path.join(mask_foldername, _item["label_filename"])
-        candidate["pseudo_label"] = os.path.join(mask_foldername, _item["pseudo_label_filename"])
-        candidate["spacing"] = _item["spacing"]
-        candidate["dim"] = _item["dim"]
-        candidate["top_region_index"] = _item["top_region_index"]
-        candidate["bottom_region_index"] = _item["bottom_region_index"]
+            # Conditionally add the label to the candidate dictionary
+            if "label_filename" in _item:
+                candidate["label"] = os.path.join(mask_foldername, _item["label_filename"])
 
-        candidates.append(candidate)
+            candidate_masks.append(candidate)
 
-    if len(candidates) == 0 and not check_spacing_and_output_size:
-        raise ValueError("Cannot find body region with given organ list.")
+    if len(candidate_masks) == 0 and not check_spacing_and_output_size:
+        raise ValueError("Cannot find body region with given anatomy list.")
```

Comment:
need to modify the unit test case

Reply:
I pushed a commit and switched to use anatomy list in `find_closest_masks`

Reply:
Thank you so much!